src/components/nabla/common/AmountSelector.tsx

```diff
@@ -65,11 +66,11 @@ export function AmountSelector<FormFieldValues extends FieldValues, TFieldName e
 
   if (onlyShowNumberInput === true) {
     return (
-      <NumberInput
+      <NumericInput
+        additionalStyle="input-ghost w-full flex-grow text-4xl font-outfit px-0 py-3"
+        register={form.register(formFieldName)}
         autoFocus
-        className="input-ghost w-full text-4xl font-outfit"
-        placeholder="Amount"
-        registerName={formFieldName}
+        maxDecimals={USER_INPUT_MAX_DECIMALS.STELLAR}
```

Comment:
The correct decimals are available at the `maxBalance.decimals` object. Let's use that value here and in l. 86. The decimals are defined per asset in this case and always using the Stellar decimals is incorrect.